store/auth.go

```diff
@@ -158,3 +160,47 @@ func ReadStoreToken() (*StoreToken, error) {
 
 	return &readStoreToken, nil
 }
+
+// RequestPackageAccessMacaroon requests a macaroon for accessing package data from the ubuntu store
+func RequestPackageAccessMacaroon() (string, error) {
+	emptyJsonData := "{}"
+	req, err := http.NewRequest("POST", myappsPackageAccessAPI, strings.NewReader(emptyJsonData))
+	req.Header.Set("accept", "application/json")
+	req.Header.Set("content-type", "application/json")
+	if err != nil {
+		return "", fmt.Errorf("cannot get package access macaroon from store: %v", err)
+	}
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	// check return code, error on 4xx and anything !200
+	switch {
+	case httpStatusCodeClientError(resp.StatusCode):
+		// unexpected client error, no user data is sent in the request
+		return "", fmt.Errorf("failed to get store token: %v", resp.Status)
+
+	case !httpStatusCodeSuccess(resp.StatusCode):
+		// unexpected server error, get Oops Id
+		oopsId := resp.Header.Get("X-Oops-Id")
+		return "", fmt.Errorf("failed to get store token: %v %v", resp.Status, oopsId)
+	}
+
+	dec := json.NewDecoder(resp.Body)
+	var responseData struct {
+		Macaroon string `json:"macaroon"`
+	}
+	if err := dec.Decode(&responseData); err != nil {
+		return "", err
+	}
+
+	macaroon := responseData.Macaroon
```

Comment:
We already have the data in a local variable. If the name is too long to be comfortable, w can make it short:
```
var r struct { ... }

if r.Macaroon == "" { ... }

return r.Macaroon
```